core/unit_test/incremental/Test12a_ThreadScratch.hpp

```diff
@@ -75,7 +75,8 @@ struct ThreadScratch {
             .set_scratch_size(scratch_level, Kokkos::PerThread(scratchSize));
 
     int max_team_size = policy.team_size_max(*this, Kokkos::ParallelForTag());
-    v                 = data_t("Matrix", pN, max_team_size);
+    ASSERT_GT(max_team_size, 0);
+    v = data_t("Matrix", pN, max_team_size);
```

Comment:
Should we add a test for the throw case? Is it reproduceable across platforms?

Reply:
I don't think there is a very consistent and backend-independent way to trigger the exceptions added here. Note that there is no requirement for the `Parallel(For|Reduce)` implementations to call `team_size_recommended` or `team_size_max` (even though we basically want the team size from `team_size_recommended` when `Kokkos::AUTO` is provided).
The host backends might just always use `team_size=1` for example.

Reply:
this change seems unnecessary.

Reply:
Right. My point was rather to demonstrate that checking is important here since switching the scratch level here silently avoided running the kernel.